### DISCUSSION TITLE
Dificuldade ao emitir NFe

### DISCUSSION THREAD

**Discussion by CaioKleiton** [TadaSoftware/PyNFe#226] [General]
# Estou com um problema ao tentar emitir a NFe... Pesquisando, cheguei até esta página e encontrei este script para a emissão de Nfe:

from xml.dom.minidom import Element
from pynfe.processamento.comunicacao import ComunicacaoSefaz
from pynfe.entidades.cliente import Cliente
from pynfe.entidades.emitente import Emitente
from pynfe.entidades.notafiscal import NotaFiscal
from pynfe.entidades.fonte_dados import _fonte_dados
from pynfe.processamento.serializacao import SerializacaoXML
from pynfe.processamento.assinatura import AssinaturaA1
from pynfe.utils.flags import CODIGO_BRASIL
from decimal import Decimal
from lxml import etree
import datetime

certificado = "certificado.pfx"
senha = '12345678'
uf = 'ac'
homologacao = True

emitente = Emitente(
    razao_social='ROTTA DO MDF LTDA',
    nome_fantasia='Rotta do MDF',
    cnpj='25164896000175',           
    codigo_de_regime_tributario='1',
    inscricao_estadual='0105007100121',
    inscricao_municipal='',
    cnae_fiscal='',           
    endereco_logradouro='Via Chico Mendes',
    endereco_numero='3915',
    endereco_bairro='Areal',
    endereco_municipio='RIO BRANCO',
    endereco_uf='AC',
    endereco_cep='76848000',
    endereco_pais=CODIGO_BRASIL
)

cliente = Cliente(
    razao_social='NFE EMITIDA EM AMBIENTE DE HOMOLOGACAO SEM VALOR FISCAL',
    tipo_documento='CPF',       
    email='[em...@email.com](https://groups.google.com/)',
    numero_documento='06171934283',
    indicador_ie=9,                 
    endereco_logradouro='Rua dos Bobos',
    endereco_numero='Zero',
    endereco_complemento='Ao lado de lugar nenhum',
    endereco_bairro='Aquele Mesmo',
    endereco_municipio='Rio Branco',
    endereco_uf='AC',
    endereco_cep='76848000',
    endereco_pais=CODIGO_BRASIL,
    endereco_telefone='11912341234',
)

nota_fiscal = NotaFiscal(
    emitente=emitente,
    cliente=cliente,
    uf=uf.upper(),
    natureza_operacao='VENDA', 
    forma_pagamento=0,         
    tipo_pagamento=1,
    modelo=55,                 
    serie='1',
    numero_nf='151515',           
    data_emissao=datetime.datetime.now(),
    data_saida_entrada=datetime.datetime.now(),
    tipo_documento=1,          
    municipio='4118402',       
    tipo_impressao_danfe=1,    
    forma_emissao='1',         
    cliente_final=1,           
    indicador_destino=1,
    indicador_presencial=1,
    finalidade_emissao='1',    
    processo_emissao='0',      
    transporte_modalidade_frete=1,
    informacoes_adicionais_interesse_fisco='Mensagem complementar',
    totais_tributos_aproximado=Decimal('21.06'),
)

nota_fiscal.adicionar_produto_servico(
    codigo='000328',                           
    descricao='NOTA FISCAL EMITIDA EM AMBIENTE DE HOMOLOGACAO SEM VALOR FISCAL',
    ncm='99999999',
    #cest='0100100',                            
    cfop='5102',
    unidade_comercial='UN',
    ean='SEM GTIN',
    ean_tributavel='SEM GTIN',
    quantidade_comercial=Decimal('12'),        
    valor_unitario_comercial=Decimal('9.75'),  
    valor_total_bruto=Decimal('117.00'),       
    unidade_tributavel='UN',
    quantidade_tributavel=Decimal('12'),
    valor_unitario_tributavel=Decimal('9.75'),
    ind_total=1,
    # numero_pedido='12345',                   
    # numero_item='123456',                    
    icms_modalidade='102',
    icms_origem=0,
    icms_csosn='400',
    pis_modalidade='07',
    cofins_modalidade='07',
    valor_tributos_aprox='21.06'
    )

nota_fiscal.adicionar_responsavel_tecnico(
    cnpj='99999999000199',
    contato='TadaSoftware',
    email='[tadaso...@gmail.com](https://groups.google.com/)',
    fone='11912341234'
  )

serializador = SerializacaoXML(_fonte_dados, homologacao=homologacao)
nfe = serializador.exportar()

a1 = AssinaturaA1(certificado, senha)
xml = a1.assinar(nfe)

con = ComunicacaoSefaz(uf, certificado, senha, homologacao)
envio = con.autorizacao(modelo='nfe', nota_fiscal=xml)

if envio[0] == 0:
    print('Sucesso!')
    print(etree.tostring(envio[1], encoding="unicode").replace('\n','').replace('ns0:','').replace(':ns0', ''))

else:
    print('Erro:')
    print(envio[1].text) # resposta
    print('Nota:')
    print(etree.tostring(envio[2], encoding="unicode")) # nfe_

# Porém, sempre que eu executo, ele está me retornando este erro abaixo:

<?xml version="1.0" encoding="utf-8"?><soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema"><soap:Body><nfeResultMsg xmlns="http://www.portalfiscal.inf.br/nfe/wsdl/NFeAutorizacao4"><retEnviNFe
versao="4.00" xmlns="http://www.portalfiscal.inf.br/nfe"><tpAmb>2</tpAmb><verAplic>SVRS202210040826</verAplic><cStat>104</cStat><xMotivo>Lote processado</xMotivo><cUF>12</cUF><dhRecbto>2022-10-08T11:15:32-05:00</dhRecbto><protNFe versao="4.00"><infProt><tpAmb>2</tpAmb><verAplic>SVRS202210040826</verAplic><chNFe>12221025164896000175550010001515151759977681</chNFe><dhRecbto>2022-10-08T11:15:32-05:00</dhRecbto><digVal>wP8Wx/L5lNErrIg2FGPAppiA3tU=</digVal><cStat>297</cStat><xMotivo>Rejeicao: Assinatura difere do calculado</xMotivo></infProt></protNFe></retEnviNFe></nfeResultMsg></soap:Body></soap:Envelope>

# Não estou conseguindo descobrir que parte do meu código está ocasionando este retorno... Se alguém puder me ajudar ficaria muito grato.

**Comment by leogregianin** [comment]
@CaioKleiton o mesmo problema ocorreu [nesse tópico](https://github.com/TadaSoftware/PyNFe/discussions/216)

Tente usar a seguintes versões das bibliotecas: signxml 2.9.0, pyOpenSSL 20.0.1 e cryptography 36.0.1

**Comment by davidncuba** [comment]
Conseguiu resolver?

**Reply by vozdatela** [reply]
Boa tarde.
Estou com o mesmo problema. Instalei as bibliotecas citadas (signxml 2.9.0, pyOpenSSL 20.0.1 e cryptography 36.0.1) mas não resolveu.

Alguma outra sugestão?
Obrigado.
